stirling-pdf/src/main/java/stirling/software/SPDF/controller/api/RemoveHeaderFooterController.java

```diff
@@ -0,0 +1,161 @@
+package stirling.software.SPDF.controller.api;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.pdfbox.multipdf.LayerUtility;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDPage;
+import org.apache.pdfbox.pdmodel.PDPageContentStream;
+import org.apache.pdfbox.pdmodel.PDPageContentStream.AppendMode;
+import org.apache.pdfbox.pdmodel.common.PDRectangle;
+import org.apache.pdfbox.pdmodel.graphics.form.PDFormXObject;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+import io.github.pixee.security.Filenames;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import stirling.software.SPDF.model.api.general.RemoveHeaderFooterForm;
+import stirling.software.common.service.CustomPDFDocumentFactory;
+import stirling.software.common.util.GeneralUtils;
+import stirling.software.common.util.WebResponseUtils;
+
+@RestController
+@RequestMapping("/api/v1/general")
+@Slf4j
+@Tag(name = "General", description = "General APIs")
+@RequiredArgsConstructor
+public class RemoveHeaderFooterController {
+
+    private final CustomPDFDocumentFactory pdfDocumentFactory;
+
+    @PostMapping(value = "/remove-header-footer", consumes = "multipart/form-data")
+    @Operation(
+            summary = "Removes headers and/or footers from a PDF document",
+            description = "Remove header and/or footer")
+    public ResponseEntity<byte[]> removeHeaderFooter(@ModelAttribute RemoveHeaderFooterForm form)
+            throws IOException {
+
+        MultipartFile pdfFile = form.getFileInput();
+
+        String pagesToDelete = form.getPages();
+        List<Integer> pagesToRemove = new ArrayList<>();
+        PDDocument sourceDoc = pdfDocumentFactory.load(pdfFile);
+        PDDocument newDoc = new PDDocument();
```

Comment:
Please use pdfDocumentFactory.createNewDocumentBasedOnOldDocument(PDDocument oldDocument)
To have metadata from first file copied over into second file so general metadata is not lost during edit!